src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -248,4 +265,116 @@ namespace AppInstaller::CLI::Workflow
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_APP_BLOCKED_BY_POLICY);
+        }
+
+        // Verifying/Acquiring product ownership
+        context.Reporter.Info() << Resource::String::MSStoreInstallTryGetEntitlement << std::endl;
+        GetEntitlementResult enr = installManager.GetFreeUserEntitlementAsync(productId, winrt::hstring(), winrt::hstring()).get();
+
+        if (enr.Status() == GetEntitlementStatus::Succeeded)
+        {
+            context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementSuccess << std::endl;
+            AICLI_LOG(CLI, Error, << "Get entitlement succeeded.");
+        }
+        else
+        {
+            if (enr.Status() == GetEntitlementStatus::NoStoreAccount)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNoStoreAccount << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement failed. No Store account.");
+            }
+            else if (enr.Status() == GetEntitlementStatus::NetworkError)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNetworkError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement failed. Network error.");
+            }
+            else if (enr.Status() == GetEntitlementStatus::ServerError)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementServerError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement succeeded. Server error. ProductId: " << Utility::ConvertToUTF8(productId));
+            }
+
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_INSTALL_FAILED);
+        }
+
+        context.Reporter.Info() << Resource::String::InstallFlowStartingPackageInstall << std::endl;
+
+        IVectorView<AppInstallItem> installItems = installManager.StartProductInstallAsync(
+            productId,              // ProductId
+            winrt::hstring(),       // CatalogId
+            winrt::hstring(),       // FlightId
+            L"WinGetCli",           // ClientId
+            false,                  // repair
+            false,
+            winrt::hstring(),
+            nullptr).get();
+
+        // We install one package at a time so the AppInstallItem list returned should contain only one entry.
+        auto installItem = installItems.GetAt(0);
+
+        auto CheckInstallStatus = [&]()
+        {
+            HRESULT errorCode = installItem.GetCurrentStatus().ErrorCode();
+            if (!SUCCEEDED(errorCode))
+            {
+                // Cancel the installation request on failure. Otherwise it'll leave an error
+                // request in Store client installation queue.
+                installItem.Cancel();
+
+                context.Reporter.Info() << Resource::String::MSStoreInstallFailed << ' ' << errorCode << std::endl;

```

Comment:
>errorCode [](start = 92, length = 9)

I don't think HRESULT automatically outputs nicely.  I would suggest adding an operator<<(HRESULT) to do so.

out << "0x" << std::hex << std::fill('0') << std::width(8) << hr;

Reply:
I created a #define so that not every long will be printed as HResult

---
In reply to: [455237189](https://github.com/microsoft/winget-cli/pull/493#discussion_r455237189) [](ancestors = 455237189)